From 94bbd051b92b5520807091574820f4eb66431964 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:25:41 -0400
Subject: [tools] perf: Fix old GCC build error in 'get_srcline'

Message-id: <1408462094-14194-173-git-send-email-jolsa@redhat.com>
Patchwork-id: 88145
O-Subject: [PATCH RHEL7.1 BZ1131394 172/325] perf tools: Fix old GCC build error in 'get_srcline'
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit a949fffb84df6f9be136198a00f796a9dc696bd0
Author: David Ahern <dsahern@gmail.com>
Date: Wed Oct 9 21:51:31 2013 -0600

description
===========
trace-event-parse.c:parse_proc_kallsyms()

    Old GCC (4.4.2) does not see through the code flow of get_srcline() and
gets confused about the status of 'file' and 'line':

    CC       /tmp/build/perf/util/srcline.o
  cc1: warnings being treated as errors
  util/srcline.c: In function "get_srcline":
  util/srcline.c:226: error: "file" may be used uninitialized in this function
  util/srcline.c:227: error: "line" may be used uninitialized in this function
  make[1]: *** [/tmp/build/perf/util/srcline.o] Error 1
  make: *** [install] Error 2
  make: Leaving directory `/home/acme/git/linux/tools/perf'
  [acme@fedora12 linux]$

Help out GCC by initializing 'file' and 'line'.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/srcline.c b/tools/perf/util/srcline.c
index 3735319..d11aefb 100644
--- a/tools/perf/util/srcline.c
+++ b/tools/perf/util/srcline.c
@@ -223,8 +223,8 @@ out:
 
 char *get_srcline(struct dso *dso, unsigned long addr)
 {
- char *file;
- unsigned line;
+ char *file = NULL;
+ unsigned line = 0;
  char *srcline;
  char *dso_name = dso->long_name;
  size_t size;
-- 
1.7.1